### PR TITLE
Fix #12028: Fix messaging when opening non-existing scores

### LINKED ISSUE #12028: [MU4 Issue] When opening a non-existing (deleted) file, you get the corruption dialog instead of missing file dialog
**To Reproduce**
1. Open score A from a Flash drive
2. Close score A
3. Eject Flash drive
4. Try to open score A from Recent scores - **you get the corruption dialog instead of missing file dialog**

### Actual result:

Capture

### Previous actual result:

**MU shows "File open error" instead of "File not found" message** 

Capture2

**Expected behavior**
MU should show message about the file is missing

**Platform information**
macOS, Windows, Linux

**Additional context**
#8884

### PR DESCRIPTION
**Pull request by cbjeukendrup**
Resolves: #12028

Closes: #12173 

Not found: 
Scherm­afbeelding 2023-07-04 om 01 07 39

No permission:
Scherm­afbeelding 2023-07-04 om 01 12 24

When an unknown error occurs, the dialog will look like this:
> **Cannot read file /Users/casper/Downloads/daft-punk-crash.mscz**
>
> An error occurred while reading this file.

But I've not been able to produce that, because I can't think of any other error.

@bkunda Please check if you agree with the copy :)

### CODE REVIEW

**Comment by bkunda** [comment]
For "not found", I'm only finding the placement of the word "currently" a bit jarring (very minor point).
I'd rather see:
> This file does not exist or cannot be accessed at the moment.

I think that's probably nicer than "...or cannot be accessed." (which might just be frustrating, especially if the user can't figure out why the file cannot be accessed). 

Otherwise, all good! Thanks again 🙏🏻

**Comment by Jojo-Schmitz** [comment]
Maybe better "Please make sure __you have__ permission to read this file"?

**Comment by cbjeukendrup** [comment]
@bkunda Fixed!

@Jojo-Schmitz On macOS, each app can be granted or denied access to individual folders. Therefore I chose this wording.

File: src/engraving/infrastructure/mscreader.cpp [hunk 1]
```diff
@@ -253,18 +254,23 @@ MscReader::ZipFileReader::~ZipFileReader()
...
 
-bool MscReader::ZipFileReader::open(IODevice* device, const path_t& filePath)
+Ret MscReader::ZipFileReader::open(IODevice* device, const path_t& filePath)
 {
     m_device = device;
     if (!m_device) {
         m_device = new File(filePath);
         m_selfDeviceOwner = true;
+
+        if (!FileInfo::exists(filePath)) {
```
Comment (RomanPudashkin) [inline src/engraving/infrastructure/mscreader.cpp:264]: You should probably check this before creating m_device

File: src/engraving/infrastructure/mscreader.cpp [hunk 2]
```diff
@@ -347,12 +353,12 @@ bool MscReader::DirReader::open(IODevice* device, const path_t& filePath)
...
     if (!FileInfo::exists(filePath)) {
         LOGD() << "not exists path: " << filePath;
-        return false;
+        return make_ret(Err::FileNotFound, filePath);
     }
 
     m_rootPath = containerPath(filePath);
 
-    return true;
+    return make_ret(Err::NoError, filePath);
```
Comment (RomanPudashkin) [inline src/engraving/infrastructure/mscreader.cpp:361]: Please use make_ok

File: src/engraving/infrastructure/mscreader.cpp [hunk 3]
```diff
@@ -407,22 +413,27 @@ ByteArray MscReader::DirReader::fileData(const String& fileName) const
...
     if (!m_device->isOpen()) {
         if (!m_device->open(IODevice::ReadOnly)) {
             LOGD() << "failed open file: " << filePath;
-            return false;
+            return make_ret(Err::FileOpenError, filePath);
         }
     }
 
-    return true;
+    return make_ret(Err::NoError, filePath);
```
Comment (RomanPudashkin) [inline src/engraving/infrastructure/mscreader.cpp:436]: Please use make_ok

**Review by RomanPudashkin — APPROVED** [review]

**Pull request merged**